core/src/main/scala/com/nvidia/spark/rapids/tool/qualification/Qualification.scala

```diff
@@ -120,6 +121,28 @@ class Qualification(outputDir: String, numRows: Int, hadoopConf: Configuration,
     }
   }
 
+  // Estimate app frequency based off of all applications in this run
+  private def estimateAppFrequency(
+    appsSum: Seq[QualificationSummaryInfo]): Seq[QualificationSummaryInfo] = {
+    var appFrequency = Map[String, Double]()
+    var windowStart: Long = Long.MaxValue
+    var windowEnd: Long = Long.MinValue
+
+    appsSum.foreach { sum =>
+      appFrequency += (sum.appName -> (1.0 + appFrequency.getOrElse(sum.appName, 0.0)))
+      windowStart = Math.min(sum.startTime, windowStart)
+      windowEnd = Math.max(windowEnd, sum.startTime + sum.estimatedInfo.appDur)
+    }
+    val windowInMonths =
+      if (windowEnd > windowStart) ((windowEnd - windowStart) / (1000.0*60*60*24*30)) else 1.0
+    appFrequency.foreach ( app =>
+      appFrequency += (app._1 -> (if (app._2 == 1) 30.0 else (app._2 / windowInMonths))))
+    appsSum.foreach { app =>
+      app.estimatedInfo.estimatedFrequency = appFrequency.getOrElse(app.appName, 30.0).round
```

Comment:
oh I see now why you made that a var.  I'm really not a fan of this approach. Arguably the frequency is something that is calculated outside of a particular application summary info because it has to look across different application runs.  I would rather see this as an add on into a completely separate field inside of the QualificationSummaryInfo.  That field can be Option and start out as None.